pyiron_contrib/atomistics/elasticity/bulk/strain.py

```diff
@@ -0,0 +1,21 @@
+import numpy as np
+
+def get_strain(structure, ref_structure, num_neighbors=None):
+    if num_neighbors is None:
+        cna = ref_structure.analyse.pyscal_cna_adaptive()
+        bulk = np.asarray([k for k in cna.keys()])[np.argmax([v for v in cna.values()])]
+        if bulk=='bcc':
+            num_neighbors = 8
+        elif bulk=='fcc' or bulk=='hcp':
+            num_neighbors = 12
+        else:
+            raise ValueError('Crystal structure not recognized')
```

Comment:
My spidey-sense is tingling... what happens if someone tries to use this on a structure with a phase boundary? i.e. there are $N$ fcc atoms and $M<N$ bcc atoms (and $L<M<N$ 'other' atoms). What are the limitations here?

Reply:
Then they will probably get a monstrous strain in the wrong area XD. But that should be fine because nothing's gonna break down. So as long as the user knows what they are doing, they can simply superpose (?) two strain fields and get the right values from each result.

Reply:
Cool, fair enough. The docstring should make clear the underlying reason that `get_strain` works this way, e.g. something to the effect that it will only be sensible if the reference is one of fcc, bcc, hcp, and that it can be trivially spatially decomposed, or similar.